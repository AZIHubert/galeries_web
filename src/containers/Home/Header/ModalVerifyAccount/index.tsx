import * as React from 'react';

interface ModalVerifyAccountI {
  loading: boolean,
  setLoading: React.Dispatch<React.SetStateAction<boolean>>
}

const ModalVerifyAccount = ({
  loading,
  setLoading,
}: ModalVerifyAccountI) => (
  <div
    data-testid="modalVerifyAccount"
  >
    <p>
      Verify your email
    </p>
    <p>
      To use Galeries, click the verification
      button in the email we sent
      to user@email.com. This helps keep
      your account secure.
    </p>
    <p>
      No email in your inbox or spam folder?
      Letâ€™s
    </p>
    <button
      disabled={loading}
      onClick={() => { if (!loading) { setLoading(true); } }}
    >
      resend it
    </button>
    <p>
      .
    </p>
  </div>
);

export default ModalVerifyAccount;
