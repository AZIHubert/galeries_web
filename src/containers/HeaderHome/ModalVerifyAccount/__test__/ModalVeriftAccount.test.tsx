import * as React from 'react';
import renderer from 'react-test-renderer';

import ThemeProvider from '#contexts/ThemeContext';

import ModalVerifyAccount from '../index';

describe('Home', () => {
  it('renders without crashing', () => {
    const mockedSetLoading = jest.fn;
    const tree = renderer.create(
      <ThemeProvider>
        <ModalVerifyAccount
          loading={false}
          setLoading={mockedSetLoading}
          currentEmail='mockedEmail@email.com'
        />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
