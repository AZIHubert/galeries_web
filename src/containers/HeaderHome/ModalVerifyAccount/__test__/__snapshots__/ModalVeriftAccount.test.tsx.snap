// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
<div
  className="sc-gsTCUz gpRCqf"
  data-testid="modalVerifyAccount"
>
  <div
    className="sc-dlfnbm iCYrgf"
  >
    <h3
      className="sc-hKgILt fVwsjm"
    >
      Verify your email
    </h3>
    <p
      data-testid="verifyAccountBody"
    >
      To use Galeries, click the verification button in the email we sent to 
      mockedEmail@email.com
      . This helps keep your account secure.
    </p>
    <div
      className="sc-jSgupP cTlbEJ"
      fontSize={1}
    >
      <p>
        No email in your inbox or spam folder? Letâ€™s 
        <span
          className="sc-eCssSg dfRMkT"
          onClick={[Function]}
        >
          resend it
        </span>
        .
      </p>
    </div>
  </div>
  <div
    className="sc-bdfBwQ kHDgwz"
  />
</div>
`;
